packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1381,7 +1401,6 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
         launchTemplateFromConfig = new ec2.LaunchTemplate(this, 'LaunchTemplate', {
           machineImage: props.machineImage,
-          keyName: props.keyName,
```

Comment:
not to introduce any breaking change here, i would suggest to not remove the property name altogether from here, props.keyPair, keep the same with passing key pair value 
          ...(props.keyName ? { keyName: props.keyName } : {}),